### DISCUSSION TITLE
Trying to render a d3d11texture2d onto an avalonia window/control

### DISCUSSION THREAD

**Discussion by sbarrac** [AvaloniaUI/Avalonia#14739] [Q&A]
I'm capturing the screen with DXGI output duplication, and i'm trying to render the output onto an Avalonia window. I managed to do this on Winforms with a basic swapchain with the window handle. [I have a minimal example here.](https://gist.github.com/sbarrac/bcf66d18375f4d99779cda4de605e658)

I looked at the gpuinterop demo and [came up with this attempt](https://gist.github.com/sbarrac/5e46cecbf9a1f4b91e4dad370db3fb93). I'm cloning the desktop texture to a shareable texture, and importing it with the gpu interop interface. No exceptions are thrown but nothing seems to happen when I call surface.updateimage. I'm hoping somebody can point me in the right direction here?

**Comment by kekekeks** [comment]
```
        PresentClonedTexture(sharedHandle, (int)desc.Width, (int)desc.Height);
        ctx->Flush();
```

You are presenting a texture when nothing is yet copied to it. Flush the context before presenting.

**Reply by kekekeks** [reply]
Try following the full example and using synchronization via keyed mutexes.
